### PR TITLE
Diagnostic contract update for the 2019-12-01-preview api version

### PR DESCRIPTION
**Pull request by promoisha**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json [hunk 1]
```diff
@@ -1784,9 +1784,59 @@
           "$ref": "#/definitions/PipelineDiagnosticSettings",
           "description": "Diagnostic settings for incoming/outgoing HTTP messages to the Backend"
         },
-        "enableHttpCorrelationHeaders": {
-          "type": "boolean",
-          "description": "Whether to process Correlation Headers coming to Api Management Service. Only applicable to Application Insights diagnostics. Default is true."
+        "httpCorrelationProtocol": {
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: >httpCorrelationProtocol [](start = 9, length = 23)

need to update the examples

**Review by solankisamir — CHANGES_REQUESTED** [review]
Need to update the examples

**Comment by akning-ms** [comment]
Hi @qiaozha  Can you help take a look the lint & breakchange fail?

**Review by ryansbenson — COMMENTED** [review]
approved from ARMs side

**Comment by qiaozha** [comment]
> Hi @qiaozha Can you help take a look the lint & breakchange fail?

@promoisha  your branch is behind Azure:master. Can you rebase it with the newest Azure:master and see if the problem still exists?

**Review by solankisamir — APPROVED** [review]
looks good

**Comment by solankisamir** [comment]
> > Hi @qiaozha Can you help take a look the lint & breakchange fail?
> 
> @promoisha your branch is behind Azure:master. Can you rebase it with the newest Azure:master and see if the problem still exists?

@qiaozha this is a forked branch off master. We will rebase the forked branch when we are ready to merge to master

**Pull request merged**